src/io/tree.cpp

```diff
@@ -219,37 +219,45 @@ double Tree::GetLowerBoundValue() const {
 
 std::string Tree::ToString() const {
   std::stringstream str_buf;
+  Common::C_stringstream(str_buf);
+
+  #if (defined(sun) || defined(__sun))
```

Comment:
Please use the following Solaris detection:
https://github.com/microsoft/LightGBM/blob/186711deae3c7c9e274467eb77f555a52ca25286/src/network/ifaddrs_patch.cpp#L6